src/wallet/wallet.cpp

```diff
@@ -1460,6 +1485,25 @@ void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRe
         // https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Wallet-Transaction-Conflict-Tracking
         SyncTransaction(tx, TxStateInactive{});
     }
+
+    const Txid& conflict_txid = tx->GetHash();
+
+    for (const CTxIn& tx_in : tx->vin) {
+        auto range =  mapTxSpends.equal_range(tx_in.prevout);
+
+        // For each wallet transaction spending this prevout...
+        for (auto _it = range.first; _it != range.second; ++_it) {
+
+            auto try_updating_state = [&](CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) {
+                // Remove the previously conflicting transaction from this wtx's entry
+                if (wtx.mempool_conflicts.erase(conflict_txid)) return TxUpdate::CHANGED;
+
+                return TxUpdate::UNCHANGED;
```

Comment:
In commit "wallet: track mempool conflicts" (a650caab9f11be7f5927f9aa556bb7d2a8ebed33)

Unimportant style point, but usually it makes more sense to return early when state is *not* changing than when it is changing:

```diff
-                if (wtx.mempool_conflicts.erase(conflict_txid)) return TxUpdate::CHANGED;
-
-                return TxUpdate::UNCHANGED;
+                if (!wtx.mempool_conflicts.erase(conflict_txid)) return TxUpdate::UNCHANGED;
+                return TxUpdate::CHANGED;
```

So if someone wants to add extra code logging or dealing with the state change, there is a place to insert it after the early return.

Alternately this could be simplified to

```c++
return wtx.mempool_conflicts.erase(conflict_txid)) ? TxUpdate::CHANGED : TxUpdate::UNCHANGED;
```

Reply:
Done